docs/src/references/c8y-configuration-management.md

```diff
@@ -0,0 +1,162 @@
+# Device Configuration Management using Cumulocity
+
+Thin-edge provides an operation plugin to
+[manage device configurations using Cumulocity](https://cumulocity.com/guides/users-guide/device-management/#to-retrieve-and-apply-a-configuration-snapshot-to-a-device-which-supports-multiple-configuration-types).
+
+* This management is bi-directional:
+  * A device can be taken as reference, all the managed files being uploaded to the cloud
+    and stored there as a configuration snapshot.
+  * A configuration snapshot can be pushed from the cloud to any devices of the same type,
+    i.e. supporting the same kind of configuration files.
+* With this operation plugin, the device owner defines the list of files
+  (usually configuration files, but not necessarily),
+  that will be managed from the cloud tenant.
+* Notably, the plugin configuration itself is managed from the cloud,
+  meaning, the device owner can update from the cloud the list of files to be managed.
+* By default, the managed files are represented on the cloud by their full path on the device.
+  Optionally, the device owner can assign an alias to each target path.  
+* When files are downloaded from the cloud to the device,
+  these files are stored in a temporary directory first.
+  They are atomically moved to their target path, only after a fully successful download.
+  The aim is to avoid breaking the system with half downloaded files.
+* Once all the files of a snapshot have been downloaded,
+  the plugin published a notification message on the local thin-edge MQTT bus.
+  The device software has to subscribe to these messages if any action is required,
+  say to check the content of file, to preprocess it or to restart a daemon. 
+* In other words, the responsibilities of the plugin are:
+  * to define the list of files under configuration management
+  * to notify the cloud when this list is updated,
```

Comment:
I think "this mechanism" of how the device admin or the device software notifies `tedge` or this `c8y_configuration_plugin` when the "supported config files list" itself is updated is not defined in this spec. Would that also be a dynamic MQTT message?

Reply:
Yes this is still pending. I was thinking to an MQTT message. But the topic and the payload have to be defined.